### PR TITLE
WIP: Additional Pandas lesson on modifying dataframes

### PR DESCRIPTION
**Pull request by colinmorris**
@dansbecker this didn't quite make it across the finish line, but it's pretty close. Thought I'd open up a PR just to see if you have any thoughts on the direction of the lesson.

If we did want to finish this up and add it to the Pandas track, remaining items would be:
- ~~Adding 1 or 2 more questions (dealing with modifying a selection and/or using `inplace=True`)~~
- ~~Fill out the final section of the reference notebook (on `inplace=True`)~~
- Complete the checking code
- Add this lesson to `pandas/track_meta.py`
- Rename "Indexing, Selecting and Assigning" to "Indexing and Selecting", and remove the brief section on assignment at the end of its reference notebook.

### CODE REVIEW

File: notebooks/pandas/raw/modifying-reference.ipynb [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+   "outputs": [],
+   "source": [
+    "reviews.head(4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In upcoming lessons, you'll see many more methods which can take an optional `inplace` argument, some of which you may already be familiar with, including:\n",
```
[the remark this answers is not in the document]
Comment (dansbecker) [reply notebooks/pandas/raw/modifying-reference.ipynb]: If that was your way of punting because you don't have time to cover these, I'll fill it in.

**Comment by dansbecker** [comment]
@colinmorris I'd have liked to see an example of arithmetic on series like `df['new_column'] = df['numerator'] / df['denominoator']` in the workbook.  I see it in the exercise.  It's common and easy, but I don't think I saw that in the reference.

I really like the reference book.  It's simultaneously short and insightful.

Also, if we leave sorting out, I think this could reasonably go in the 4th slot in the course, before grouping and sorting.  You are closer to the Pandas course than I am though... what do you think?

File: notebooks/pandas/raw/modifying-reference.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (colinmorris) [reply notebooks/pandas/raw/modifying-reference.ipynb]: Oh, no these are methods that are actually covered in later existing pandas lessons. ("later" under the assumption that this lesson would slot in between "summary functions and maps" and "grouping and sorting"). The mapping from lessons to these methods is:

4. grouping and sorting

- reset_index
- sort_values
- sort_index

5. dtypes and missing data

- fillna
- Series.replace

6. renaming combining

- rename
- rename_axis

**Comment by dansbecker** [comment]
Sorry. I think we'd discussed this lesson going before grouping and
sorting, and I'd just forgotten that.

**Comment by colinmorris** [comment]
> I'd have liked to see an example of arithmetic on series like df['new_column'] = df['numerator'] / df['denominoator'] in the workbook. I see it in the exercise. It's common and easy, but I don't think I saw that in the reference.

Doing arithmetic on series to get new series is covered a lot in the previous lesson ("Summary functions and maps"). The reference notebook also has a couple examples of setting a column to a transformed version of a column (just not arithmetically transformed):

```python
reviews['COUNTRY'] = reviews.country.str.upper()
```

So I'm inclined to say this isn't necessary to include, but not something I feel strongly about.

**Pull request closed without merging**